packages/storybook-addon-featureflags/src/switch.js

```diff
@@ -0,0 +1,73 @@
+import * as glamor from 'glamor'
+import React, {useState} from 'react'
+import PropTypes from 'prop-types'
+
+import {layout} from '@pluralsight/ps-design-system-core'
+import { blue, utility, contrast } from '@pluralsight/ps-design-system-core/js/colors.2020.js'
+
+const styles = {
+  container: _ =>
+    glamor.css({
+      border: `1px solid ${utility[40]}`,
+      borderRadius: 4,
+      color: 'inherit',
+      cursor: 'pointer',
+      display: 'inline-block',
+      overflow: 'hidden',
+      padding: 0,
+      margin: layout.spacingXSmall,
+      textAlign: 'left',
+      width: 150,
+      wordWrap: 'break-word'
+    }),
+  preview: checked => glamor.css({
+      height: 60,
+      background: checked  ? blue.base : utility[20]
+  }),
+  info: _ =>
+    glamor.css({
+      borderTop: `1px solid ${utility[40]}`,
+      color: contrast.textIconLightBgHigh,
+      padding: '0 10px'
+    }),
+  title: _ =>
+    glamor.css({
+      color: 'inherit',
+      fontWeight: 'bold',
+      fontSize: 12,
+      padding: `${layout.spacingXSmall} 0`,
+      textTransform: 'uppercase'
+    }),
+  input: _ => 
+    glamor.css({
+      position: 'absolute',
+      width: 0,
+      height: 0,
+      opacity: 0,
+      left: 0,
+      top: 0,
+    })
+}
+
+export const Switch = ({ onChange, name, ...props }) =>{
+  const [checked, setChecked] = useState()
+  function handleChange(evt) {
+    setChecked(evt.target.checked)
+    onChange(evt, {[name]: evt.target.checked})
+  }
+
+  return (
+    <label  {...styles.container()} >
+      <input onChange={handleChange} checked={checked} type="checkbox" {...styles.input()}/>
```

Comment:
For some reason, clicking this is not affecting the Button appearance for me.

![2020-01-08 15 15 50](https://user-images.githubusercontent.com/552044/72020841-e7aacd80-3229-11ea-9823-fcd3fb57a411.gif)

Reply:
In the recording, I don't get an error.  I refreshed and tried again and go this:

```
backend.js:6 Warning: A component is changing an uncontrolled input of type checkbox to be controlled. Input elements should not switch from uncontrolled to controlled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components
    in input (created by Switch)
    in label (created by Switch)
    in Switch (created by FeatureFlagsPanel)
    in div (created by FeatureFlagsPanel)
    in FeatureFlagsPanel
    in div (created by Context.Consumer)
    in Styled(div)
    in div (created by Context.Consumer)
    in Styled(div)
    in Unknown
    in Unknown
    in Unknown
    in Unknown (created by Context.Consumer)
    in ManagerConsumer (created by _default)
    in _default (created by Layout)
    in div (created by Context.Consumer)
    in Styled(div) (created by Panel)
    in Panel (created by Layout)
    in div (created by Context.Consumer)
    in Styled(div) (created by Main)
    in div (created by Context.Consumer)
    in Styled(div) (created by Main)
    in Main (created by Layout)
    in Layout (created by Context.Consumer)
    in WithTheme(Layout)
    in Unknown
    in Unknown
    in div (created by Context.Consumer)
    in Styled(div)
    in Unknown
    in Unknown (created by SizeMeRenderer(Component))
    in SizeMeReferenceWrapper (created by SizeMeRenderer(Component))
    in SizeMeRenderer(Component) (created by SizeMe(Component))
    in SizeMe(Component) (created by Manager)
    in ThemeProvider (created by Manager)
    in Manager (created by Context.Consumer)
    in Location (created by QueryLocation)
    in QueryLocation (created by Root)
    in LocationProvider (created by Root)
    in HelmetProvider (created by Root)
    in Root
```

Reply:
I hate that error but easy fix. 

As for it not doing anything I was waiting on hover values I had a test class in there earlier now I'm going to apply the correct colors.